src/Keter/Main.hs

```diff
@@ -55,74 +64,93 @@ import qualified Filesystem.Path as FP (FilePath)
+
+-- | Load up Keter config and evaluate a ReaderT context with it
+runKeterConfigReader :: MonadIO m
+                     => FilePath
+                     -> ReaderT KeterConfig m a
+                     -> m a
+runKeterConfigReader input ctx = do
+    exists <- liftIO $ doesFileExist input
+    config <- liftIO $
         if exists
             then do
                 eres <- decodeFileRelative input
                 case eres of
                     Left e -> throwIO $ InvalidKeterConfigFile input e
                     Right x -> return x
             else return defaultKeterConfig { kconfigDir = input }
-    f config
-
-withLogger :: FilePath
-           -> (KeterConfig -> (LogMessage -> IO ()) -> IO a)
-           -> IO a
-withLogger fp f = withConfig fp $ \config -> do
-    mainlog <- LogFile.openRotatingLog
-        (kconfigDir config </> "log" </> "keter")
-        LogFile.defaultMaxTotal
-
-    f config $ \ml -> do
-        now <- getCurrentTime
-        let bs = encodeUtf8 $ T.pack $ concat
-                [ take 22 $ show now
-                , ": "
-                , show ml
-                , "\n"
-                ]
-        LogFile.addChunk mainlog bs
-
-withManagers :: FilePath
-             -> [FilePath -> IO Plugin]
-             -> (KeterConfig -> HostMan.HostManager -> AppMan.AppManager -> (LogMessage -> IO ()) -> IO a)
-             -> IO a
-withManagers input mkPlugins f = withLogger input $ \kc@KeterConfig {..} log -> do
-    processTracker <- initProcessTracker
-    hostman <- HostMan.start
-    portpool <- PortPool.start kconfigPortPool
-    tf <- TempFolder.setup $ kconfigDir </> "temp"
-    plugins <- mapM ($ kconfigDir) mkPlugins
+    runReaderT ctx config
+
+-- | Running the Keter logger requires a context with access to a KeterConfig, hence the
+-- MonadReader constraint. This is versatile: 'runKeterConfigReader', or use the free 
+-- ((->) KeterConfig) instance.
+runKeterLogger :: (MonadReader KeterConfig m, MonadIO m, MonadUnliftIO m)
+               => LoggingT m a
+               -> m a
+runKeterLogger ctx = do
+    KeterConfig{..} <- ask
+    let logName = kconfigDir </> "log" </> "keter.log"
+    let logType = 
+          if kconfigRotateLogs
+            then FL.LogFile (LogFile.defaultRotationSpec logName) LogFile.defaultBufferSize
+            else FL.LogStderr LogFile.defaultBufferSize
+    withRunInIO $ \rio -> FL.withFastLogger logType $ \logger ->
+        rio $ runLoggingT ctx (formatLog logger)
+    where
+        formatLog log loc _ lvl msg = do
+            now <- liftIO getCurrentTime
+            -- Format: "$time|$module$:$line_num|$log_level> $msg"
+            let bs = FL.toLogStr $ encodeUtf8 $ T.pack $ concat
+                    [ take 22 $ show now
+                    , "|"
+                    , show (L.loc_module loc) <> ":" <> show (fst $ L.loc_start loc)
+                    , "|"
+                    , drop 3 $ show lvl
+                    , "> "
+                    , show msg
```

Comment:
calling show on msg causes it to be quoted, this is a bit reeudant:

![image](https://user-images.githubusercontent.com/3874017/229065336-22027db3-4d5f-4f9d-ab67-3974bca8f4fa.png)


keep in mind that LogStr is a monoid, 
and you can use toLogStr to convert ordinary strings to log message, no need for a Text represnetation in between.

Reply:
I also meant to change that to a `drop 5`

Reply:
so to clarify, I think it'd be better to convert everything toLogStr as soon as you can and concat on that.
looks like module name is also quoted.

Reply:
Formatting should be good now.